test/e2e_flags.go

```diff
@@ -50,13 +49,6 @@ func initializeFlags() *EnvironmentFlags {
 	f.ClientConfig.InitFlags(flag.CommandLine)
 	f.TestClientConfig.InitFlags(flag.CommandLine)
 
-	// We want to do this defaulting for tests only. The flags are reused between tests
-	// and production code and we want to make sure that production code defaults to
-	// the in-cluster config correctly.
-	if f.Kubeconfig == "" {
-		f.Kubeconfig = clientcmd.RecommendedHomeFile
```

Comment:
doesn't this mean that this will no longer properly default the flag when running locally?

Reply:
So, checking the code, I think this should work, as the fallback is precisely this path as well.

Reply:
yeah, looks like the fallback was added after this code, which explains why it was added, and why it's now redundant 🤞